guides/source/upgrading_ruby_on_rails.md

```diff
@@ -257,6 +257,35 @@ See the [Testing Rails Applications](https://guides.rubyonrails.org/testing.html
 
 If you run a single file's tests (`bin/rails test test/models/user_test.rb`), `test:prepare` will not run before it.
 
+### Import syntax from `@rails/ujs` is modified
+
+Starting from Rails 7.1, the syntax for importing modules from `@rails/ujs` is modified.
+Rails no longer supports the direct import
+of a module from `@rails/ujs`.
+If users create and build a JavaScript file
+with the line `import { fileInputSelector } from "@rails/ujs"`,
+they will encounter the error `export 'fileInputSelector'
+(imported as 'fileInputSelector')
+was not found in '@rails/ujs' (possible exports: default)`.
+
+In Rails 7.1, users should first import the Rails object directly from `@rails/ujs`.
+Users can then import specific modules
+from the Rails object.
```

Comment:
There is some rather peculiar line-wrapping going on here 👀

Reply:
Good point! I have modified the PR to fix the line-wrapping.